R/get_cdc_data.R

```diff
@@ -0,0 +1,41 @@
+#' Retrieve CDC growth chart metadataset
+#'
+#' Retrieves CDC growth chart metadataset
+#'
+#' @param type Type of Chart
+#'
+#'   *Permitted Values*: `"height"`, `"weight"`, `"bmi"`
+#'
+#' @return A data-frame of the appropriate growth chart from the CDC data
+#'
+#' @details
+#'
+#' Each data frame will be denoted as below:
+#' \describe{
+#'   \item{\code{SEX}}{double sex where 1 = male, 2 = female}
+#'   \item{\code{AGE}}{double age in months}
+#'   \item{\code{L}}{double Box-Cox transformation for normality}
+#'   \item{\code{M}}{double Median}
+#'   \item{\code{S}}{double Coefficient of variation}
+#' }
+#'
+#' from [CDC website](https://www.cdc.gov/growthcharts/percentile_data_files.htm)
+#'
+#' @family metadata
+#' @keywords metadata
+#'
+#' @export
+#'
+#' @examples
+#' get_cdc_data(type = "bmi") |> head()
+get_cdc_data <- function(type) {
+  # Limit input values to the three datasets available from CDC data
+  assert_character_scalar(type, values = c("height", "weight", "bmi"))
+
+  # Grab the appropriate one
+  switch(type,
+    height = cdc_htage,
+    weight = cdc_wtage,
+    bmi = cdc_bmiage
```

Comment:
@rossfarrugia the tribble statements are basically wrapped into here, say the user calls:

`get_cdc_data(type = "bmi")` 

the switch function will be parsed until line 39, where the return object is `cdc_bmiage` which is an internally stored dataset, which was created with the `cdc_metadata_creation.R`

Reply:
> @rossfarrugia the tribble statements are basically wrapped into here, say the user calls:
> 
> `get_cdc_data(type = "bmi")`
> 
> the switch function will be parsed until line 39, where the return object is `cdc_bmiage` which is an internally stored dataset, which was created with the `cdc_metadata_creation.R`

@Pierre-Wallet